tests/B4RealTest.spec.ts

```diff
@@ -136,11 +165,172 @@ describe("B4REAL token tests", async () => {
+      ethers.utils.formatEther(await B4REAL.balanceOf(taxAddress))
+    );
+
+    // no tax should be applied so full amount should be sent to secondUserAddress
+    // and the tax address should have no balance change
+    expect(currentUser2Balance.toString()).to.equal(amount.toString());
+    expect(openingTaxBalance.toString()).to.equal(closingTaxBalance.toString());
+  });
+
+  it("Should test the transferFrom function and apply tax correctly", async () => {
+    await B4REAL.includeInFee(secondUserAddress);
+    const isWhitelisted = await B4REAL.whitelisted(secondUserAddress);
+    expect(isWhitelisted).to.equal(true);
+
+    const openingBalance = await B4REAL.balanceOf(secondUserAddress);
+
+    // Just the tokens from the previous test in this account
+    expect(openingBalance.toString()).to.equal(ethers.utils.parseEther("1"));
+
+    const openingTaxBalance = Number(
+      ethers.utils.formatEther(await B4REAL.balanceOf(taxAddress))
+    );
+
+    // Just the tokens from the previous test in this account
+    expect(openingTaxBalance).to.equal(10);
+
+    const amount = ethers.utils.parseEther("1");
+
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [userAddress]
+    });
+
+    // Approve secondUserAddress
+    await B4REAL.connect(userSigner).approve(secondUserAddress, amount);
+
+    const allowance = await B4REAL.allowance(userAddress, secondUserAddress);
+
+    // Check allowance is correct, including call from before
+    expect(allowance.toString()).to.equal(amount.toString());
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [userAddress]
+    });
+
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [secondUserAddress]
+    });
+
+    await B4REAL.connect(secondUserSigner).transferFrom(
+      userAddress,
+      secondUserAddress,
+      amount
+    );
+    await hre.network.provider.request({
+      method: "hardhat_stopImpersonatingAccount",
+      params: [secondUserAddress]
+    });
+
+    const currentUser2Balance = await B4REAL.balanceOf(secondUserAddress);
+
+    const closingTaxBalance = Number(
+      ethers.utils.formatEther(await B4REAL.balanceOf(taxAddress))
+    );
+
+    // tax should be applied
+    expect(currentUser2Balance.toString()).to.equal("1900000000000000000");
+    expect(closingTaxBalance.toString()).to.equal("10.1");
+  });
+
+  it("Should allow admin to change the tax address", async () => {
+    const deployerAddress = await deployer.getAddress();
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [deployerAddress]
+    });
+    try {
+      await B4REAL.updateB4REALTaxAddress(
+        "0xe3F078F80A530cCD3BbF221612dDca3B0724579D"
+      );
+    } catch (error) {
+      expect(getRevertMessage(error)).to.equal(
+        "New address cannot be the same"
+      );
+    }
```

Comment:
Dont do this.  Use .to.be.revertedWith https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#revert-with-message

Reply:
I worked with Barry on this. For reasons we are still not sure about using `revertedWith` was breaking other tests